### PR TITLE
Add Other checks new controls and update the structure

### PR DESCRIPTION
**Pull request by khushboo9024**
### Checklist
- [ ] Issue(s) linked

1. [x] compute_vm_data_and_os_disk_uses_managed_disk
2. [x] compute_vm_scale_set_automatic_upgrade_enabled
3. [x] synapse_workspace_data_exfil_protection_enabled
4. [x] signalr_service_no_free_tier_sku
5. [x] redis_cache_min_tls_1_2
6. [x] securitycenter_pricing_standard
7. [x] mariadb_server_ssl_enabled
8. [x] kubernetes_cluster_logging_enabled
9. [x] kubernetes_cluster_key_vault_secret_rotation_enabled
10. [x] kubernetes_cluster_max_pod_50
11. [x] kubernetes_cluster_network_policy_enabled
12. [x] appservice_web_app_always_on
13. [x] appservice_plan_minimum_sku
14. [x] appservice_web_app_health_check_enabled
15. [x] search_service_uses_managed_identity
16. [x] search_service_replica_count_3
17. [x] container_instance_container_group_in_virtual_network
18. [x] compute_vm_scale_set_ssh_key_authentication_linux
19. [x] kusto_cluster_sku_with_sla
20. [x] eventgrid_topic_local_auth_enabled
21. [x] eventgrid_topic_identity_provider_enabled
22. [x] sql_server_threat_detection_all_enabled
23. [x] monitor_log_profile_retention_365_days
24. [x] postgres_db_server_latest_tls_version

### CODE REVIEW

File: other_checks/redis.sp [hunk 1]
```diff
@@ -0,0 +1,17 @@
+locals {
+  other_redis_common_tags = merge(local.azure_compliance_common_tags, {
+    service = "Azure/Redis"
+  })
+}
+
+benchmark "other_redis" {
+  title = "Redis"
+  children = [
+    control.azure_redis_cache_in_virtual_network,
```
Comment (rajlearner17) [inline other_checks/redis.sp:10]:
```suggestion
    control.redis_cache_in_virtual_network,
```
@khushboo9024 Should we remove the work azure from this to be in naming consistency?

**Review by rajlearner17 — APPROVED** [review]
LGTM after validating the logic and functionality of the queries.

File: regulatory_compliance/search.sp [hunk 2]
```diff
@@ -45,6 +45,26 @@ control "search_service_uses_private_link" {
   })
 }
 
+control "search_service_uses_managed_identity" {
+  title       = "Cognitive Search services should use managed identity"
+  description = "Use a managed identity for enhanced authentication security."
```
Comment (madhushreeray30) [inline regulatory_compliance/search.sp:50]:
```suggestion
  description = "Cognitive Search services should use a managed identity for enhanced authentication security."
```

File: regulatory_compliance/search.sp [hunk 3]
```diff
@@ -174,3 +194,46 @@ query "search_service_uses_private_link" {
...
+  sql = <<-EOQ
+    select
+      s.id as resource,
+      case
+        when identity ->> 'type' = 'SystemAssigned' then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when identity ->> 'type' = 'SystemAssigned' then name || ' uses managed identity.'
+        else name || ' not uses managed identity.'
```
Comment (madhushreeray30) [inline regulatory_compliance/search.sp:208]:
```suggestion
        else name || ' not use managed identity.'
```

File: regulatory_compliance/sql.sp [hunk 4]
```diff
@@ -143,6 +143,16 @@ control "sql_server_auditing_storage_account_destination_retention_90_days" {
   })
 }
 
+control "sql_server_threat_detection_all_enabled" {
+  title       = "SQL servers threat detection should be enabled for all"
+  description = "This control ensures that SQL server threat detection is enabled for all."
```
Comment (madhushreeray30) [inline regulatory_compliance/sql.sp:148]:
```suggestion
  title       = "SQL server threat detection should be enabled for all"
  description = "This control ensures that SQL server threat detection is enabled for all."
```

File: regulatory_compliance/redis.sp [hunk 5]
```diff
@@ -147,3 +157,28 @@ query "redis_cache_no_basic_sku" {
...
+      c.id as resource,
+      case
+        when minimum_tls_version is null then 'alarm'
+        when minimum_tls_version = '1.2' then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when minimum_tls_version is null then c.name || ' minimum TLS version not set.'
+        when minimum_tls_version = '1.2' then c.name || ' minimum TLS version set to ' || minimum_tls_version || '.'
+        else c.name || ' minimum TLS version set to ' || minimum_tls_version || '.'
```
Comment (madhushreeray30) [inline regulatory_compliance/redis.sp:173]:
```suggestion
        else c.name || ' minimum TLS version set to ' || minimum_tls_version || '.'
```

File: regulatory_compliance/postgres.sp [hunk 6]
```diff
@@ -375,3 +385,25 @@ query "postgres_db_server_allow_access_to_azure_services_disabled" {
...
+query "postgres_db_server_latest_tls_version" {
+  sql = <<-EOQ
+    select
+      s.id as resource,
+      case
+        when minimal_tls_version = 'TLS1_2' then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when minimal_tls_version = 'TLS1_2' then name || ' using the latest version of TLS encryption.'
```
Comment (madhushreeray30) [inline regulatory_compliance/postgres.sp:397]:
```suggestion
        when minimal_tls_version = 'TLS1_2' then name || ' uses the latest version of TLS encryption.'
```

File: regulatory_compliance/kubernetes.sp [hunk 7]
```diff
@@ -541,3 +581,105 @@ query "kubernetes_cluster_upgrade_channel" {
...
+    )
+    select
+      c.id as resource,
+      case
+        when n.id is not null then 'alarm'
+        else 'ok'
+      end as status,
+      case
+        when n.id is not null then c.name || ' nodes have pods less than 50.'
+        else c.name || ' nodes have pods greater than 50.'
```
Comment (madhushreeray30) [inline regulatory_compliance/kubernetes.sp:650]:
```suggestion
        when n.id is not null then c.name || ' nodes have less than 50 pods.'
        else c.name || ' nodes have greater than 50 pods.'
```

File: other_checks/other.sp [hunk 8]
```diff
@@ -8,45 +8,27 @@ benchmark "other" {
...
+    benchmark.other_eventgrid,
+    benchmark.other_keyvault,
+    benchmark.other_mariadb,
+    benchmark.other_monitor,
+    benchmark.other_postgres,
+    benchmark.other_redis,
+    benchmark.other_securitycenter,
+    benchmark.other_signalr,
+    benchmark.other_sql,
+    benchmark.other_synapse,
```
Comment (madhushreeray30) [inline other_checks/other.sp:31]:
```suggestion
    benchmark.other_synapse
```

**Review by madhushreeray30 — COMMENTED** [review]
Please have a look at the comments, Thanks!

**Review by madhushreeray30 — APPROVED** [review]

File: regulatory_compliance/eventgrid.sp [hunk 9]
```diff
@@ -137,3 +157,45 @@ query "eventgrid_domain_identity_provider_enabled" {
...
+  sql = <<-EOQ
+    select
+      a.id as resource,
+      case
+        when disable_local_auth then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when disable_local_auth then a.name || ' local authentication disabled.'
+        else a.name || ' local authentication enabled.'
```
Comment (misraved) [inline regulatory_compliance/eventgrid.sp:171]: As per the description of the control, I think the condition needs to be reversed, i.e. if local authentication is disabled, then it should be in alarm.

**Review by misraved — COMMENTED** [review]
Please take a look at the review comments. Thanks!!

**Review by misraved — APPROVED** [review]

**Pull request merged**